src/ir/properties.cpp

```diff
@@ -43,6 +43,12 @@ static bool isValidInConstantExpression(Module& wasm, Expression* expr) {
     return true;
   }
 
+  if (auto* refAs = expr->dynCast<RefAs>()) {
+    if (refAs->op == ExternExternalize || refAs->op == ExternInternalize) {
+      return isValidInConstantExpression(wasm, refAs->value);
+    }
+  }
```

Comment:
We shouldn't need to recurse here because recursion should be handled by `isValidConstantExpression`.